statsmodels/sandbox/mice/mice.py

```diff
@@ -0,0 +1,1167 @@
+            creating the histogram for observed values.
+        all_hist_args : dict
+            Keyword arguments to be passed to pyplot.hist when
+            creating the histogram for all values.
+
+        Returns:
+        --------
+        The matplotlib figure on which the histograms were drawn
+        """
+
+        from statsmodels.graphics import utils as gutils
+        from matplotlib.colors import LinearSegmentedColormap
+
+        if ax is None:
+            fig, ax = gutils.create_mpl_ax(ax)
+        else:
+            fig = ax.get_figure()
+
+        ax.set_position([0.1, 0.1, 0.7, 0.8])
+
+        ixm = self.ix_miss[col_name]
+        ixo = self.ix_obs[col_name]
+
+        imp = self.data[col_name].iloc[ixm]
+        obs = self.data[col_name].iloc[ixo]
+
+        for di in imp_hist_args, obs_hist_args, all_hist_args:
+            if 'histtype' not in di:
+                di['histtype'] = 'step'
+
+        ha, la = [], []
+        if len(imp) > 0:
+            h = ax.hist(np.asarray(imp), **imp_hist_args)
+            ha.append(h[-1][0])
+            la.append("Imp")
+        h1 = ax.hist(np.asarray(obs), **obs_hist_args)
+        h2 = ax.hist(np.asarray(self.data[col_name]), **all_hist_args)
+        ha.extend([h1[-1][0], h2[-1][0]])
+        la.extend(["Obs", "All"])
+
+        leg = fig.legend(ha, la, 'center right', numpoints=1)
+        leg.draw_frame(False)
+
+        ax.set_xlabel(col_name)
+        ax.set_ylabel("Frequency")
+
+        return fig
+
+    def perturb_bootstrap(self, vname):
+        """
+        Perturbs the model's parameters using a bootstrap.
+        """
+
+        endog_obs, exog_obs, exog_miss =\
+                   self.get_split_data(vname)
+
+        m = len(endog_obs)
+        rix = np.random.randint(0, m, m)
+        endog_boot = endog_obs[rix]
+        exog_boot = exog_obs[rix, :]
+        klass = self.model_class[vname]
+        self.models[vname] = klass(endog_boot, exog_boot,
+                                  **self.init_args)
+        self.results[vname] = self.models[vname].fit(**self.fit_args)
+        self.params[vname] = self.results[vname].params
+
+    def perturb_gaussian(self, vname):
+        """
+        Perturbs the model's parameters by sampling from the Gaussian
+        approximation to the sampling distribution of the parameter
+        estimates.  Optionally, the scale parameter is perturbed by
+        sampling from its asymptotic Chi^2 sampling distribution.
+        """
+
+        endog_obs, exog_obs, exog_miss =\
+                   self.get_split_data(vname)
+
+        klass = self.model_class[vname]
+        formula = self.conditional_formula[vname]
+        self.models[vname] = klass.from_formula(formula,
+                                            self.data,
+                                            **self.init_args[vname])
+        self.results[vname] = self.models[vname].fit(
+            **self.fit_args[vname])
```

Comment:
question: Do we have to call this repeatedly for the same fit? in that case we would have redundant fit for generating rvs.
I guess, No, in this case where we will have new (imputed) `data` in each round.